src/test/scala/com/cloudera/sparkts/models/ARIMAXSuite.scala

```diff
@@ -0,0 +1,136 @@
+  val tsTest_2 = new BreezeDenseVector(Array(100.0 ,98 ,102 ,98 ,112 ,99 ,99 ,87 ,103 ,115 ,101 ,125 ,117 ,109 ,111 ,105))
+  val xregTest_2 = new DenseMatrix(rows = 16, cols = 4, data = Array( 465,453,472,454,432,431,475,393,437,537,462,539,471,455,466,490,
+    1,1,0,0,0,0,0,1,1,0,0,0,0,0,1.0,1,
+    24,24,25,25,25,25,25,25,25,23,23,23,23,23,23,23,
+    51,54,49,46,42,41,45,46,48,41,42,48,43,47,48,46 ))
+
+  def mean(t: BreezeDenseVector[Double]): Double = {
+    val ts = new DenseVector(t.toArray)
+    var sum = 0.0
+    ts.values.map( va => sum += va)
+    sum/t.length
+  }
+  def maxDeviation(mean: Double, ts: BreezeDenseVector[Double]): Double ={
+    var max = 0.0
+    for (t <- ts){
+      val error = Math.abs(t - mean)
+      if (error > max) max = error }
+    max
+  }
+
+  /* First data set */
+  test("MAX(0,0,1) 1 true - first data set"){
+    val model = ARIMAX.fitModel(0, 0, 1, tsTrain, xregTrain, 1)
+    assert(model.coefficients.length == 6)
+    val results = model.predict(tsTest, xregTest)
+    results.length should be (tsTest.length)
+
+    for ( i <- results ){
+      i should be (mean(tsTest) +- 10)
+    }
+  }
+
+  test("MAX(0,0,2) 1 true"){
+    val model = ARIMAX.fitModel(0, 0, 2, tsTrain_2, xregTrain_2, 1)
+    assert(model.coefficients.length == 11)
+    val results = model.predict(tsTest_2, xregTest_2)
+    results.length should be (tsTest_2.length)
+
+    for ( i <- results ){
+      i should be (mean(tsTest_2) +- 2)
+    }
+  }
+  test("IMAX(0,1,1) 1 true"){
+    val model = ARIMAX.fitModel(0, 1, 1, tsTrain_2, xregTrain_2, 1)
+    assert(model.coefficients.length == 10)
+    val results = model.predict(tsTest_2, xregTest_2)
+    results.length should be (tsTest_2.length)
+
+    for ( i <- results ){
+      i should be (mean(tsTest_2) +- 2)
+    }
+  }
+
+  test("ARIMAX(2,1,1) 1 true false - first data set"){
+    val model = ARIMAX.fitModel(2, 1, 1, tsTrain, xregTrain, 1, includeIntercept = false)
+    assert(model.coefficients.length == 8)
+    val results = model.predict(tsTest, xregTest)
+    results.length should be (tsTest.length)
+    
+    // The timeseries data are quite scattered that's why we multiply max deviation by 2 
+    for ( i <- results ){
+      i should be (mean(tsTest) +- maxDeviation(mean(tsTest),tsTest )*2)
+    }
+  }
+
+  test("ARIMAX(1,1,1) 1 true true"){
+    val model = ARIMAX.fitModel(1, 1, 1, tsTrain_2, xregTrain_2, 1)
+    assert(model.coefficients.length == 11)
+    val results = model.predict(tsTest_2, xregTest_2)
+    results.length should be (tsTest_2.length)
+    
+    for ( i <- results ){
+      i should be (mean(tsTest_2) +- 5)
+    }
+  }
+
+  test("ARIMAX(2,1,1) 1 true false"){
+    val model = ARIMAX.fitModel(2, 1, 1, tsTrain_2, xregTrain_2, 1, includeIntercept = false)
+    assert(model.coefficients.length == 12)
+    val results = model.predict(tsTest_2, xregTest_2)
+    results.length should be (tsTest_2.length)
+
+    for ( i <- results ){
```

Comment:
Nit (applies to other places throughout the test): spacing should be `for (i <- results) {`